packages/std/src/math/decimal.rs

```diff
@@ -17,6 +18,9 @@ use super::{Uint128, Uint256};
 #[derive(Copy, Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, JsonSchema)]
 pub struct Decimal(#[schemars(with = "String")] Uint128);
 
+#[derive(Debug, PartialEq)]
+pub struct DecimalRangeExceeded;
```

Comment:
That's cool that we're using a custom error type! Could we implement `std::error::Error` for this? `Result` technically allows any type as an error type, but it's useful to make sure that type actually implements `Error` for the sake of things like converting to an `anyhow::Error` or holding it in a `Box<dyn Error>`.

One easy way would probably be to derive `thiserror::Error`, since we're already importing it.